### PR TITLE
Refactor/pn-13039: refactor 2nd step "recipient" in notification creation

### PR DESCRIPTION
**Pull request by fbianchicodermine**
## Short description
Refactor of UI second tab of manual notification according to the mockup.

## List of changes proposed in this pull request
- Remove "at" field in address

## How to test
Start pa-webapp. Create manual notification and check 2nd tab

### CODE REVIEW

File: packages/pn-pa-webapp/src/components/NewNotification/Recipient.tsx [hunk 1]
```diff
@@ -146,52 +135,31 @@ const Recipient: React.FC<Props> = ({
         .test('taxIdDependingOnRecipientType', t('fiscal-code-error'), function (value) {
           return taxIdDependingOnRecipientType(value, this.parent.recipientType);
         }),
-      digitalDomicile: yup.string().when('showDigitalDomicile', {
-        is: true,
-        then: requiredStringFieldValidation(tc, 320).matches(dataRegex.email, t('pec-error')),
```
Comment (AndreaCimini90) [inline packages/pn-pa-webapp/src/components/NewNotification/Recipient.tsx:151]: why did you remove the requiredStringFieldValidation?

File: packages/pn-pa-webapp/src/components/NewNotification/Recipient.tsx [hunk 2]
```diff
@@ -387,112 +305,68 @@ const Recipient: React.FC<Props> = ({
...
-                          errors,
-                          index,
-                          values,
-                          setFieldTouched,
-                          setFieldValue
-                        )
+              <Fragment key={recipient.id}>
+                <FormBox testid="RecipientFormBox">
+                  {/* Soggetto giuridico */}
+                  <Stack display="flex" direction="row" justifyContent="space-between">
```
Comment (AndreaCimini90) [inline packages/pn-pa-webapp/src/components/NewNotification/Recipient.tsx:318]: remove display flex

File: packages/pn-pa-webapp/src/components/NewNotification/Recipient.tsx [hunk 3]
```diff
@@ -387,112 +305,68 @@ const Recipient: React.FC<Props> = ({
...
+                            setFieldTouched,
+                            setFieldValue
+                          )
+                        }
+                      >
+                        <Delete color="error" />
+                      </ButtonNaked>
+                    )}
+                  </Stack>
+                  <FormControl sx={{ marginTop: 3, marginBottom: 1 }}>
```
Comment (AndreaCimini90) [inline packages/pn-pa-webapp/src/components/NewNotification/Recipient.tsx:338]: use mt and mb

File: packages/pn-pa-webapp/src/components/NewNotification/Recipient.tsx [hunk 4]
```diff
@@ -505,152 +379,129 @@ const Recipient: React.FC<Props> = ({
                         errors={errors}
                         setFieldValue={setFieldValue}
                         handleBlur={handleBlur}
-                        width={12}
+                        sx={{ marginTop: 2 }}
```
Comment (AndreaCimini90) [inline packages/pn-pa-webapp/src/components/NewNotification/Recipient.tsx:382]: use mt

File: packages/pn-pa-webapp/src/components/NewNotification/Recipient.tsx [hunk 5]
```diff
@@ -505,152 +379,129 @@ const Recipient: React.FC<Props> = ({
...
                       />
-                      {paymentMode !== PaymentModel.NOTHING && (
-                        <>
+                    </Grid>
+                    <Grid item lg={4} sx={{ display: { xs: 'none', lg: 'block' } }}></Grid>
```
Comment (AndreaCimini90) [inline packages/pn-pa-webapp/src/components/NewNotification/Recipient.tsx:385]: why this empty grid item?

File: packages/pn-pa-webapp/src/components/NewNotification/Recipient.tsx [hunk 6]
```diff
@@ -505,152 +379,129 @@ const Recipient: React.FC<Props> = ({
...
-                          name={`recipients[${index}].showPhysicalAddress`}
-                          data-testid={`showPhysicalAddress${index}`}
-                          label={`${t('add-physical-domicile')}*`}
                         />
                       </Grid>
+                    )}
+                  </Grid>
+
+                  {/* Indirizzo */}
+                  <Box marginTop={4} marginBottom={3}>
```
Comment (AndreaCimini90) [inline packages/pn-pa-webapp/src/components/NewNotification/Recipient.tsx:428]: use mt and mb

File: packages/pn-pa-webapp/src/components/NewNotification/Recipient.tsx [hunk 7]
```diff
@@ -505,152 +379,129 @@ const Recipient: React.FC<Props> = ({
...
+                      setFieldValue={setFieldValue}
+                      touched={touched}
+                      errors={errors}
+                      recipient={index}
+                      handleBlur={handleBlur}
+                    />
+                  </Grid>
+
+                  {/* Domicilio digitale */}
+                  <Box marginTop={4} marginBottom={2}>
```
Comment (AndreaCimini90) [inline packages/pn-pa-webapp/src/components/NewNotification/Recipient.tsx:443]: use mt and mb

File: packages/pn-pa-webapp/src/components/NewNotification/Recipient.tsx [hunk 8]
```diff
@@ -505,152 +379,129 @@ const Recipient: React.FC<Props> = ({
...
-                        >
-                          {t('add-recipient')}
-                        </ButtonNaked>
-                      </Stack>
-                    )}
+                    </Grid>
+                  )}
+                </FormBox>
+                {values.recipients.length < 5 && values.recipients.length - 1 === index && (
+                  <Stack mt={2} display="flex" direction="row" justifyContent="space-between">
```
Comment (AndreaCimini90) [inline packages/pn-pa-webapp/src/components/NewNotification/Recipient.tsx:489]: remove display flex

File: packages/pn-pa-webapp/src/components/NewNotification/__test__/Recipient.test.tsx [hunk 9]
```diff
@@ -92,39 +84,36 @@ const testRecipientFormRendering = async (
...
-    const address = physicalForm?.querySelector(
-      `input[name="recipients[${recipientIndex}].address"]`
-    );
-    expect(address).toHaveValue(recipient.address);
+  expect(physicalForm).toBeInTheDocument();
+  const address = physicalForm?.querySelector(
+    `input[name="recipients[${recipientIndex}].address"]`
+  );
+
+  if (recipient) {
```
Comment (AndreaCimini90) [inline packages/pn-pa-webapp/src/components/NewNotification/__test__/Recipient.test.tsx:95]: why this if?

**Review by AndreaCimini90 — CHANGES_REQUESTED** [review]

File: packages/pn-pa-webapp/src/components/NewNotification/Recipient.tsx (hunk 1 above)
Comment (fbianchicodermine) [reply packages/pn-pa-webapp/src/components/NewNotification/Recipient.tsx:151]: Because now "digitalDomicile" is an input always displayed but not required

File: packages/pn-pa-webapp/src/components/NewNotification/Recipient.tsx (hunk 5 above)
Comment (fbianchicodermine) [reply packages/pn-pa-webapp/src/components/NewNotification/Recipient.tsx]: To respect input width across various screen sizes. But I removed it and added a new grid

File: packages/pn-pa-webapp/src/components/NewNotification/__test__/Recipient.test.tsx (hunk 9 above)
Comment (fbianchicodermine) [reply packages/pn-pa-webapp/src/components/NewNotification/__test__/Recipient.test.tsx:92]: Because in function `testRecipientFormRendering` recipient is optional.
